common/src/main/java/org/valkyrienskies/mod/mixin/feature/ai/node_evaluator/SwimNodeEvaluatorMixin.java

```diff
@@ -0,0 +1,188 @@
+package org.valkyrienskies.mod.mixin.feature.ai.node_evaluator;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.PathNavigationRegion;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.level.pathfinder.NodeEvaluator;
+import net.minecraft.world.level.pathfinder.PathComputationType;
+import net.minecraft.world.level.pathfinder.SwimNodeEvaluator;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.valkyrienskies.mod.common.VSGameUtilsKt;
+import org.valkyrienskies.mod.common.config.VSGameConfig;
+
+@Mixin(SwimNodeEvaluator.class)
+public abstract class SwimNodeEvaluatorMixin extends NodeEvaluator {
+
+    //region Single block obstacle path type
+    @Redirect(
+        at = @At(value = "INVOKE",
+            target = "Lnet/minecraft/world/level/BlockGetter;getFluidState(Lnet/minecraft/core/BlockPos;)Lnet/minecraft/world/level/material/FluidState;"),
+        method = "getBlockPathType(Lnet/minecraft/world/level/BlockGetter;III)Lnet/minecraft/world/level/pathfinder/BlockPathTypes;"
+    )
+    private FluidState getFluidStateRedirectPathType(final BlockGetter instance, final BlockPos blockPos) {
+        final FluidState[] fluidState = {instance.getFluidState(blockPos)};
+        Level level = null;
+        if (!VSGameConfig.SERVER.getAiOnShips()) {
+            if (instance instanceof PathNavigationRegion) {
+                level = ((PathNavigationRegionAccessor) instance).getLevel();
+            } else if (instance instanceof Level) {
+                level = (Level) instance;
+            }
+            if (level != null && fluidState[0].isEmpty()) {
+
+                final double origX = blockPos.getX();
+                final double origY = blockPos.getY();
+                final double origZ = blockPos.getZ();
+                final Level finalLevel = level;
+                VSGameUtilsKt.transformToNearbyShipsAndWorld(level, origX,
+                    origY, origZ, 1,
```

Comment:
Why are you using `transformToNearbyShipsAndWorld` if it only uses the last x,y,z that it gets and ignores the rest?